packages/contracts/contracts/deploy/PayableDAppFactory.sol

```diff
@@ -133,16 +140,16 @@ contract PayableDAppFactory is
 
         // transfer tokens from sender to this contract
         require(
-            token.transferFrom(_msgSender, address(this), _cost),
+            token.transferFrom(_msgSender, payee, _cost),
             "PaymentDAppFactory: failed to transfer tokens"
         );
 
         uint256 currentRunway = runway[_dapp];
 
         // if runway was in the past, update to now
-        currentRunway = currentRunway < block.timestamp
-            ? block.timestamp
-            : currentRunway;
+        if (currentRunway < block.timestamp) {
+            currentRunway = block.timestamp;
+        }
```

Comment:
I don't understand why are you updating `currentRunway` if `< block.timestamp`.
Couldn't you call `extendRunway(dapp, 0)`, which would update the `currentRunway` to `block.timestamp`, and then later call `extendRunway` again with some actual value, saving you potentially a lot of tokens?

Reply:
Suppose I pay for 1 week.
After 1 week the app stops.

1 month later I extend for 1 week.
I'm not paying for the 3 weeks it was stopped.
It's 1 week from now.

Reply:
Ohh, now I get it now. Yeah, that makes sense.